services/order.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+from django.utils.dateparse import parse_datetime
+
+from db.models import Ticket, Order
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str | None = None
+) -> Order:
+
+    user = get_user_model().objects.get(username=username)
+
+    # with transaction.atomic():
+    order = Order.objects.create(user=user)
+    if date:
+        order.created_at = parse_datetime(date)
```

Comment:
you can pass date as is - no need for conversion here. Don't forget to clean up imports as well